src/app/appointment/appointment.component.ts

```diff
@@ -193,18 +193,35 @@ export class AppointmentComponent {
   }
 
   /**
-   * Converts UTC hour to local hour.
+   * Converts EST/EDT datetime by hour to UTC
+   */
+  getUtcHour(hour: number) {
+    return moment.tz(this.printHour(hour), 'HH:mm', 'America/Toronto').tz('UTC').format('HH:mm');
+  }
+
+  /**
+   * Converts EST/EDT hour to local hour.
    * @param  {number} hour EST/EDT hour
    * @return {string}      Local hour
    */
   getLocalHour(hour: number): string {
-    let timezone = this.getLocalTimezone();
+    if (!this.localTimezone) {
+      this.localTimezone = this.getLocalTimezone();
+    }
 
-    // create moment in EST/EDT timezone
+    // read time for EST/EDT timezone
     let eastern = moment.tz(this.printHour(hour), 'HH:mm', 'America/Toronto');
+    let local = eastern.clone().tz(this.localTimezone);
+
+    // check if appointment falls to the next day
```

Comment:
Does this also cover the correct allocation to the week day column in the table?